### PR TITLE
CORTX-28919:RGW: Add rgw & motr parameters which are not exposed to Gconf

### PR DESCRIPTION
**Pull request by sachitanands**
Signed-off-by: Sachitanand Shelake <sachitanand.shelake@seagate.com>

Updating rgw config files with additional keys which are for RGW & motr parameters.
https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/915276544/CORTX+Config+Keys+for+RGW 
# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by sachitanands** [comment]
testing changes using below jobs,
docker images : https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/5091/

File: src/setup/rgw.py [hunk 1]
```diff
@@ -202,18 +203,22 @@ def upgrade(conf: MappedConf):
...
-                'consul http endpoint is not specified in the conf.'
-                f' Listed endpoints: {endpoints}')
-        # Relace 'http' with 'consul' and port - 8500 in endpoint string.
+        http_endpoints = Rgw._fetch_endpoint_url(conf, const.CONSUL_ENDPOINT_KEY, 'http')
         consul_fqdn = http_endpoints[seq].split(':')[1]
         consul_url = 'consul:' + consul_fqdn + ':8500'
         return consul_url
 
+    @staticmethod
+    def _fetch_endpoint_url(conf: MappedConf, confstore_endpoint_key, endpoint_type):
```
Comment (shriya-deshmukh) [inline src/setup/rgw.py:212]: _fetch_endpoint_url(conf: MappedConf, confstore_endpoint_key: str, endpoint_type: str):

File: src/setup/rgw.py [hunk 2]
```diff
@@ -573,31 +577,53 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
...
+        Log.info(f'adding paramters to {client_section} in {rgw_config_file}')
+
+        # e.g config_key_mapping = [[confstore_key1, actual_rgw_config_key1],
+        # [confstore_key2, actual_rgw_config_key2], ..]
+        for confstore_key, config_key in config_key_mapping:
+            # fetch actual value of parameter from confstore.
+            config_value = Conf.get(Rgw._rgw_conf_idx, confstore_key)
+            if not config_value :
+                raise SetupError(errno.EINVAL,
+                    f'Confstore key/value is missing for key {confstore_key}')
```
Comment (shriya-deshmukh) [inline src/setup/rgw.py:624]: You can use '_get_cortx_conf' function to read key value from gconf.
config_value= Rgw._get_cortx_conf(conf, confstore_key)

**Review by shriya-deshmukh — CHANGES_REQUESTED** [review]

File: src/setup/rgw.py [hunk 3]
```diff
@@ -573,31 +577,53 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
...
         backend_store = Conf.get(Rgw._rgw_conf_idx, 'client>rgw backend store')
-        if not backend_store in SUPPORTED_BACKEND_STORES:
+        if not backend_store in const.SUPPORTED_BACKEND_STORES:
             raise SetupError(errno.EINVAL,
-                f'Supported rgw backend store are {SUPPORTED_BACKEND_STORES},'
+                f'Supported rgw backend store are {const.SUPPORTED_BACKEND_STORES},'
                 f' currently configured one is {backend_store}')
+
+    @staticmethod
+    def _update_rgw_config(conf: MappedConf, client_section: string, config_key_mapping: list):
```
[the remark this answers is not in the document]
Comment (sachitanands) [reply src/setup/rgw.py]: done

File: src/setup/rgw.py (hunk 1 above)
Comment (sachitanands) [reply src/setup/rgw.py]: done

File: src/setup/rgw.py (hunk 2 above)
Comment (sachitanands) [reply src/setup/rgw.py]: done

File: src/rgw/const.py [hunk 4]
```diff
@@ -46,6 +46,39 @@
 SSL_CERT_CONFIGS = {"country" : "IN", "state" : "MH", "locality" : "Pune",
     "organization" : "Seagate Technology", "CN" : "seagate.com"}
 SSL_DNS_LIST = [u'*.seagate.com', u'localhost', u'*.localhost']
+SSL_CERT_PATH_KEY = 'cortx>common>security>ssl_certificate'
+RGW_ENDPOINT_KEY =  f'cortx>{COMPONENT_NAME}>service>endpoints'
```
Comment (cdeshmukh) [inline src/rgw/const.py:50]: Avoid RGW in constants & variables in any new code, as name may change it later. 
We are driving in other PRs to replace RGW references with generic, lets not add it here again

File: src/rgw/setup/rgw.py [hunk 5]
```diff
@@ -373,18 +377,18 @@ def _update_rgw_config_with_endpoints(conf: MappedConf, endpoints: dict, instanc
...
         ssl_port = 8443
         ssl_port = ssl_port + (instance - 1)
-        ssl_cert_path = Rgw._get_cortx_conf(conf, 'cortx>common>security>ssl_certificate')
+        ssl_cert_path = Rgw._get_cortx_conf(conf, const.SSL_CERT_PATH_KEY)
         Conf.set(
             Rgw._rgw_conf_idx,
-            f'client.rgw-{instance}>{ADMIN_PARAMETERS["RGW_FRONTENDS"]}',
-            f'beast port={port} ssl_port={ssl_port} ssl_certificate={ssl_cert_path} ssl_private_key={ssl_cert_path}')
+            f'client.rgw-{instance}>{const.ADMIN_PARAMETERS["RGW_FRONTENDS"]}',
+            f'beast port={port} ssl_port={ssl_port} ssl_certificate={ssl_cert_path}, ssl_private_key={ssl_cert_path}')
```
Comment (shriya-deshmukh) [inline src/rgw/setup/rgw.py:384]: Please remove comma (',')  - `f'beast port={port} ssl_port={ssl_port} ssl_certificate={ssl_cert_path} ssl_private_key={ssl_cert_path}'`

File: src/rgw/setup/rgw.py [hunk 6]
```diff
@@ -573,31 +576,53 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
...
+        svc_config_file = os.path.join(svc_config_dir, const.RGW_CONF_FILE)
+        Rgw._load_rgw_config(Rgw._rgw_conf_idx, f'ini://{svc_config_file}')
+        Log.info(f'adding paramters to {client_section} in {svc_config_file}')
+
+        # e.g config_key_mapping = [[confstore_key1, actual_svc_config_key1],
+        # [confstore_key2, actual_svc_config_key2], ..]
+        for confstore_key, config_key in config_key_mapping:
+            # fetch actual value of parameter from confstore.
+            config_value = Rgw._get_cortx_conf(conf, confstore_key)
+            if not config_value :
```
Comment (shriya-deshmukh) [inline src/rgw/setup/rgw.py:621]: we can remove this if condition, As this check is already there in _get_cortx_conf() fun, 
```
config_value = Rgw._get_cortx_conf(conf, confstore_key)
Conf.set(Rgw._rgw_conf_idx, f'{client_section}>{config_key}', {config_value})
```

File: src/rgw/setup/rgw.py (hunk 5 above)
Comment (sachitanands) [reply src/rgw/setup/rgw.py]: done

File: src/rgw/const.py (hunk 4 above)
Comment (sachitanands) [reply src/rgw/const.py]: done. renamed variable/ method names from rgw - > svc

File: src/rgw/setup/rgw.py (hunk 6 above)
Comment (sachitanands) [reply src/rgw/setup/rgw.py]: done

**Review by shriya-deshmukh — APPROVED** [review]

File: src/rgw/setup/rgw.py [hunk 7]
```diff
@@ -237,35 +241,35 @@ def _get_rgw_config_path(conf: MappedConf):
...
+        auth_secret = Rgw._get_cortx_conf(conf, f'cortx>{const.COMPONENT_NAME}>auth_secret')
         err_str = f'user: {user_name} exists'
         # decrypt secret key.
         try:
             cluster_id = Rgw._get_cortx_conf(conf, 'cluster>id')
-            cipher_key = Cipher.gen_key(cluster_id, DECRYPTION_KEY)
+            cipher_key = Cipher.gen_key(cluster_id, const.DECRYPTION_KEY)
             password = Cipher.decrypt(cipher_key, auth_secret.encode('utf-8'))
             password = password.decode('utf-8')
         except CipherInvalidToken as e:
```
Comment (rohit-k-dwivedi) [inline src/rgw/setup/rgw.py:275]: this change was intended in this PR https://github.com/Seagate/cortx-rgw-integration/pull/44/files#r820619861  

`create_usr_cmd = f'sudo radosgw-admin user create --uid={user_name} --access-key \` 
to
`create_usr_cmd = f'radosgw-admin user create --uid={user_name} --access-key \`

Comment (sachitanands) [reply src/rgw/setup/rgw.py:275]: done

**Comment by sachitanands** [comment]
testing these changes using,
docker image : https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/5163/console 
 cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-5163-custom-ci
 cortx-docker.colo.seagate.com/seagate/cortx-all:2.0.0-5163-custom-ci
deployment job : https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/408/console

**Review by cdeshmukh — APPROVED** [review]

**Comment by sachitanands** [comment]
testing these changes using,
docker image : 
      https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/5176/console
      cortx-docker.colo.seagate.com/seagate/cortx-all:2.0.0-5176-custom-ci
      cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-5176-custom-ci
Deployment job : https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/461/

**Comment by sachitanands** [comment]
Deployment and IO testing went ahead with above changes & docker iamges..
No issues observed and config file was properly updated with all config parameters..
image

**Pull request merged**